docs/Coding/mapgen.rst

```diff
@@ -0,0 +1,569 @@
+the tiles at other temperatures avoid it.
+
+The last step to generate the terrain is to fill in the gaps between the ocean
+and the hills. This is done according to terrain fractions that depend on the
+global ``wetness`` and ``temperature`` settings. Terrains are generated in
+patches, according to properties defined in ``terrain.ruleset`` and conditions
+on the tile. The following combinations are generated one at a time:
+
+.. table:: Terrain produced by the generator and their matching to tiles
+  :widths: auto
+  :align: center
+
+  +--------+-----------+-----------+----------+---------+-------------+---------+-----------+
+  |        | Terrain properties               | Tile properties                 |           |
+  +        +-----------+-----------+----------+---------+-------------+---------+           +
+  | Label  | Required  | Preferred | Avoided  | Wetness | Temp.       | Height  | Thr.      |
+  +========+===========+===========+==========+=========+=============+=========+===========+
+  | Forest | Foliage   | Temperate | Tropical | All     | Not frozen  | \-      | 60        |
+  +--------+-----------+-----------+----------+---------+-------------+---------+-----------+
+  | Jungle | Foliage   | Tropical  | Cold     | All     | Tropical    | \-      | 50        |
+  +--------+-----------+-----------+----------+---------+-------------+---------+-----------+
+  | Swamp  | Wet       | \-        | Foliage  | Not dry | Hot         | Low     | 50        |
+  +--------+-----------+-----------+----------+---------+-------------+---------+-----------+
+  | Desert | Dry       | Tropical  | Cold     | Dry     | Not frozen  | Not low | 80        |
+  +--------+-----------+-----------+----------+---------+-------------+---------+-----------+
+  | Desert | Dry       | Tropical  | Wet      | All     | Not frozen  | Not low | 40        |
+  +--------+-----------+-----------+----------+---------+-------------+---------+-----------+
+  | Ice    | Frozen    | \-        | Mountain | \-      | \-          | \-      | \-        |
+  +--------+-----------+-----------+----------+---------+-------------+---------+-----------+
+  | Tundra | Cold      | \-        | Mountain | \-      | \-          | \-      | \-        |
+  +--------+-----------+-----------+----------+---------+-------------+---------+-----------+
+  | Plains | Temperate | Green     | Mountain | \-      | \-          | \-      | \-        |
+  +--------+-----------+-----------+----------+---------+-------------+---------+-----------+
+
+Terrain patches expand outwards from a seed tile until the required tile
+properties are no longer met or a threshold in colatitude and height difference
+is reached (*Thr.* in the table). Ice, tundra, and plains/grassland are
+generated to fill in gaps and do not expand in patches.
+
+The algorithm to match the desired terrain properties to the ruleset-defined
+terrain types by first collecting all terrains with the required property. Then,
```

Comment:
s/terrains/terrain

Reply:
Here I mean terrain "types" from the ruleset again.